pkgs/by-name/ap/apacheAnt/package.nix

```diff
@@ -16,66 +18,29 @@ stdenv.mkDerivation rec {
     sha256 = "1l8say86bz9gxp4yy777z7nm4j6m905pg342li1aphc14p5grvwn";
   };
 
+  postPatch = ''
+    # Hard-code immutable paths
+    substituteInPlace bin/ant \
+      --replace-fail '$JAVA_HOME' '${jre.home}' \
```

Comment:
Do we actually need jre in buildInputs, if we just substitute it like this?
(haven't checked)

Reply:
Also, will this change make it so that we don't actually need to add both jdk and ant into nativeBuildInputs when using ant?

Reply:
I don't think so. I did it because that's what the convention is, as far as I understand it; happy to remove it if that isn't the case.

Reply:
Also, will this change make it so that we don't actually need to add both jdk and ant into nativeBuildInputs when using ant?

Also also, `jdk` vs `jre`? (I'd say `jdk`)

Reply:
From my testing, if we use `jdk`, that is the case, but setting it to `jdk` makes it difficult to use a different version of Java without overriding `ant`.

Reply:
Is there a reason that we shouldn't have consumers explicitly specify a JDK?